### ISSUE TITLE
System Volume Slider

### ISSUE THREAD

**Issue by kvosbur** [Aayush9029/Rifi#8]
I don't know how os independent this implementation would be, but I think it might be preferable to have the system volume control be a slider from 0 - 100 instead of being an up or down. I found a Stack Overflow post that describes how this might happen, but I don't know which OS it would work for. https://stackoverflow.com/questions/20828752/python-change-master-application-volume

**Comment by Aayush9029** [comment]
**Volume sider would definitely be elegant.**

Here's my plan.

1. For this to work we'll have to first determine user's OS. 
2. Depending on OS we'll have to find a way to get system volume information and send it to javascript via python. (let's say PC's volume is set at 45%, we'll have to send the value to web server)
3. Then parse the info using javascript and change the initial value of slider accordingly. (So we'll set the value of the slider to 45% instead of some arbitrary value) 
4. And finally an event listener will report changes to the slider values and we will change the volume accordingly. (If slider value increases by 3% we'll increase system volume by 3%)

Here are some issues:
- Pyautogui doesn't seem to work with MacOs however using apple script we might be able to fix that.
- We'll have to find a way of determining current volume information in linux and windows or we could set the volume to 50% at launch and set our slider's volume to 50% so we can skip **step 2**

And since volume buttons doesn't work on MacOs we'll have to find a work around for that.

*I'll start prototyping and tell you where it goes. Feel free to issue a pull request if you happen to find a easier way of implementing it.*

TLDR; calibrating system volume and slider value might be an issue however a work around might be to set a default volume at launch.

**Comment by Aayush9029** [comment]
This will not be worked on

*I don't have much time to work and debug*

**Comment by kvosbur** [comment]
I have found some documentation on how to set (and I might need to figure out get) system volume for windows, linux, and macos. I really want this feature to be added in so I will put the time in to get it working, though I will not be able to test it on macos.

**Comment by Aayush9029** [comment]
I’ll test for macOS. Good luck.

**Comment by Aayush9029** [comment]
> Slider works flawlessly on macOS. 

> Some minor ui changes has been implemented the "volume info" is hidden when not in use. 


*Forward and Rewind button doesn't seem to work on macOS (10.15.3 \19D76\)*

*I'll look into that soon*
